src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -62,10 +110,11 @@ public static void WriteParameterNullChecks(this CodeWriter writer, IReadOnlyCol
             {
                 if (parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue))
                 {
+                    var defaultValue = parameter.DefaultValue.Value;
                     writer.Append($"{parameter.Name} ??= ");
-                    if (TypeFactory.IsStruct(parameter.Type) || parameter.DefaultValue.Value.Type.Equals(parameter.Type))
+                    if (defaultValue.IsNewInstanceSentinel || TypeFactory.IsStruct(parameter.Type) || parameter.DefaultValue.Value.Type.Equals(parameter.Type))
```

Comment:
I expected WriteConstant to handle `IsNewInstanceSentinel`

Reply:
We don't get into `WriteConstant` we this isn't checked explicitly.

Reply:
This hack is here to handle the `new Uri(...)` conversion might be good to move it to WriteConversion eventually.